sw/host/spiflash/spiflash.cc

```diff
@@ -160,12 +160,14 @@ bool ParseArgs(int argc, char **argv, SpiFlashOpts *options) {
   assert(options);
   const struct option long_options[] = {
       {"input", required_argument, nullptr, 'i'},
-      {"dev-id", required_argument, nullptr, 'd'},
-      {"dev-sn", required_argument, nullptr, 'n'},
-      {"dump-frames", required_argument, nullptr, 'x'},
-      {"verilator", required_argument, nullptr, 's'},
+      {"dev-id", optional_argument, nullptr, 'd'},
+      {"dev-sn", optional_argument, nullptr, 'n'},
+      {"dump-frames", optional_argument, nullptr, 'x'},
+      {"verilator", optional_argument, nullptr, 's'},
       {"help", no_argument, nullptr, 'h'},
       {nullptr, no_argument, nullptr, 0}};
+  // default to FTDI
+  options->action = SpiFlashAction::kFtdi;
```

Comment:
Maybe this should be in the definition of `SpiFlashOpts` instead?

Reply:
I don't think so because an alternative program could have a different default (the use of FTDI here is to match the old behaviour from before the new options, not inherent to the SpiFlashAction). The current definition of SpiFlashOpts defaults to an error value that the caller has not set this field and I think that is correct.

Reply:
Thinking some more (now I'm not in a rush to push a program to my FPGA) this should probably be replaced by adding to the case 'i' below
if (spi_flash_options.action == SpiFlashAction::kInvalid) {
    // default to FTDI
    options->action = SpiFlashAction::kFtdi;
}

That way if there is no input file given (the other sense of required argument) then the action will not be set and the error will be raised.
@moidx would you agree?

Reply:
Probably after the switch case block?

```c++
if (!options->input.empty() && spi_flash_options.action == SpiFlashAction::kInvalid) {
  options->action = SpiFlashAction::kFtdi;
}
```

Reply:
Turns out the return is from within the case so put there.